src/squlearn/encoding_circuit/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,577 @@
+    (e.g. mycircuit.draw("mpl", decompose = True)).
+
+    Args:
+        num_qubits (int): Number of qubits of the qcnn encoding circuit.
+        num_features (int): Dimension of the feature vector. By default this is 0,
+            so an feature map must be provided (or can be changed to x by set_num_features(x)).
+            If num_features is bigger then 0, then in the get_circuit function a ZFeatureMap is build.
+        default (bool): If True, the default circuit is build.
+    """
+
+    def __init__(self, num_qubits: int = 0, num_features=0, default=False) -> None:
+        super().__init__(num_qubits, num_features)
+        self._num_parameters = 0  # counts the number of parameters used
+        self._left_qubits = [
+            i for i in range(num_qubits)
+        ]  # stores, how many qubits can be controlled yet
+        self._operations_list = []  # stores the operations applied in the qcnn
+        self._default = default
+        if default:
+            if num_qubits == 0:
+                print("To generate a default circuit provide a number of qubits > 0.")
+            else:
+                self.default_circuit()
+
+    @property
+    def num_parameters(self) -> int:
+        """Returns the number of trainable parameters of the current encoding circuit."""
+        return self._num_parameters
+
+    @property
+    def left_qubits(self) -> list:
+        """Returns the qubits which one can operate on in the current circuit as list."""
+        return self._left_qubits
+
+    @property
+    def operations_list(self) -> list:
+        """Returns the list of operators currently acting on the encoding circuit."""
+        return self._operations_list
+
+    def set_num_qubits(self, n: int = 0):
+        """Set the number of qubits of the circuit."""
+        if n > 0:
+            if self.num_qubits == n:
+                print("The number of qubits did not change.")
+            else:
+                self._num_qubits = n
+                self._left_qubits = [i for i in range(n)]
+                self._num_parameters = 0
+                if self._default:
+                    self._operations_list = []
+                    self.default_circuit()
+                else:
+                    for operation in self.operations_list:
+                        if operation[0] == "C":
+                            self.convolution(*operation[1:], new_operation=False)
+                        elif operation[0] == "P":
+                            self.pooling(*operation[1:], new_operation=False)
+                        elif operation[0] == "F":
+                            self.fully_connected(*operation[1:], new_operation=False)
+                            break
+
+    def set_num_features(self, n: int = 0):
+        """Change the number of features in the feature map of the circuit."""
+        if n > 0:
+            self._num_features = n
+
+    def convolution(
+        self,
+        QC="None",
```

Comment:
```suggestion
        quantum_circuit=None,
```
for style purposes, let's change `QC` to `quantum_circuit`. This also applies to all later occurences of `QC`. Also change `"None"` to `None`